docs/design/persistent-storage.md

```diff
@@ -0,0 +1,235 @@
+# Persistent Stores
+
+This document proposes a model for managing persistent, cluster-scoped storage for applications requiring long lived data.
+
+### tl;dr
+
+`PersistentStore` objects contain a VolumeSource with a lifecycle greater than the pods it is used by.  Many types of VolumeSources can be implemented as volume plugins, with each implementing an interface for use within a controller.  A pod with persistence requirements (specifically pod.spec.volumes.volume.persistenceRequirements) is bound to a matching VolumeSource from a pool of available stores.  PersistentStore objects give that VolumeSource longevity.
```

Comment:
I would love to see this called a NetVolume or, if you really think it will one day encompass local disk, PersistentVolume or something similar.  Way more obvious.